sentry-ruby/lib/sentry/propagation_context.rb

```diff
@@ -40,31 +40,37 @@ def initialize(scope, env = nil)
       @baggage = nil
       @incoming_trace = false
 
-      if env
-        sentry_trace_header = env["HTTP_SENTRY_TRACE"] || env[SENTRY_TRACE_HEADER_NAME]
-        baggage_header = env["HTTP_BAGGAGE"] || env[BAGGAGE_HEADER_NAME]
-
-        if sentry_trace_header
-          sentry_trace_data = self.class.extract_sentry_trace(sentry_trace_header)
-
-          if sentry_trace_data
-            @trace_id, @parent_span_id, @parent_sampled = sentry_trace_data
-
-            @baggage = if baggage_header && !baggage_header.empty?
-                        Baggage.from_incoming_header(baggage_header)
-                      else
-                        # If there's an incoming sentry-trace but no incoming baggage header,
-                        # for instance in traces coming from older SDKs,
-                        # baggage will be empty and frozen and won't be populated as head SDK.
-                        Baggage.new({})
-                      end
-
-            @baggage.freeze!
-            @incoming_trace = true
-          end
+      # Invoking code could pass a nil env, so let's ||= it so it's easier to work with.
+      env ||= {}
+
+      # Trace string could be passed from the invoking code,
+      # or via the environment variable.
+      sentry_trace_string = env["HTTP_SENTRY_TRACE"] || env[SENTRY_TRACE_HEADER_NAME] || ENV["SENTRY_TRACE"]
```

Comment:
I may be wrong, but I don't think the "environment variables" here means the system environment variables? It likely means request envs too?